src/app/singletons/battle_object_world.rs

```diff
@@ -105,9 +105,9 @@ impl BattleObjectWorld {
         }
     }
 
-    pub fn move_speed(&self) -> *const phx::Vector3f {
+    pub fn move_speed<'a>(&'a self) -> &'a phx::Vector3f {
```

Comment:
This shouldn't need an `<'a>`. AFAIK, the rust compiler can infer the lifetime of the vector based off of the lifetime of self. I think clippy lints would inform you of this, so maybe run `cargo skyline clippy` on the repo locally to confirm (or maybe don't since I need to add some global flags to ignore the `_x` naming scheme I have...)